src/app/shell/personal-cabinet/provider/create-achievement/create-achievement.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Subject } from 'rxjs';
+import { ActivatedRoute, Params } from '@angular/router';
+import { takeUntil } from 'rxjs/operators';
+import { UserWorkshopService } from 'src/app/shared/services/workshops/user-workshop/user-workshop.service';
+import { Workshop } from 'src/app/shared/models/workshop.model';
+import { FormControl, Validators } from '@angular/forms';
+import { Store } from '@ngxs/store';
+import { MatDialog } from '@angular/material/dialog';
+import { Achievement } from 'src/app/shared/models/achievement.model';
+import { Constants } from 'src/app/shared/constants/constants';
+import { ConfirmationModalWindowComponent } from 'src/app/shared/components/confirmation-modal-window/confirmation-modal-window.component';
+import { ModalConfirmationType } from 'src/app/shared/enum/modal-confirmation';
+import { CreateAchievement } from 'src/app/shared/store/user.actions';
+import { Teacher } from 'src/app/shared/models/teacher.model';
+import { Child } from 'src/app/shared/models/child.model';
+
+@Component({
+  selector: 'app-create-achievement',
+  templateUrl: './create-achievement.component.html',
+  styleUrls: ['./create-achievement.component.scss'],
+})
+export class CreateAchievementComponent implements OnInit, OnDestroy {
+  workshop: Workshop;
+  teachers: Teacher[];
+  children: Child[];
+  workshopId: string;
+  title;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+
```

Comment:
I believe, it should be a formgroup?) not just values? Please, consider the following existed approach. create a formgroup that has such form controls